Soleras3/react-challenge-project-jan-2023
# Feature: Route Guarding

## Changes
1. Create `GuardedRoute` that wraps a react route with an authentication check.
2. Change order and view-orders routes to `GuardedRoute`

## Purpose
Users that aren't logged in shouldn't be able to view or make orders. See: https://github.com/Shift3/react-challenge-project-jan-2023/issues/6

## Approach
This puts the authentication check into one logical place (routing). I prefer not to extend or wrap system components but it's required with react router v5 (see Learning).

## Testing Steps
Without logging in, navigate directly to `/order` or `/view-orders`. Observe they redirect to login.

## Learning
There are numerous resources on creating guarded / private routes with React Router, including [their own docs](https://v5.reactrouter.com/web/example/auth-workflow).

Future refactoring note: Preferring composition over extension, v6 [offers a cleaner solution](https://medium.com/@dennisivy/creating-protected-routes-with-react-router-v6-2c4bbaf7bc1c). 

Closes https://github.com/Shift3/react-challenge-project-jan-2023/issues/6
